AK/NumberFormat.cpp

```diff
@@ -109,4 +110,28 @@ DeprecatedString human_readable_digital_time(i64 time_in_seconds)
     return builder.to_deprecated_string();
 }
 
+DeprecatedString human_readable_integer(i64 number)
+{
+    bool is_negative = false;
+    if (number < 0) {
+        is_negative = true;
+        number *= -1;
+    }
+
+    Vector<u64> separated_pieces;
```

Comment:
Maybe give this some inline storage since most numbers will only be split into a few parts. 
```suggestion
    Vector<u64, 10> separated_pieces;
```

Reply:
If we go the "reduce allocation" way, you can even get rid of the vector.

Not tested, this has issues for sure:
```cpp
auto const abs_number = abs(number);
auto const number_of_decimal = log10(abs_number);

StringBuilder builder;
builder.ensure_capacity(number_of_decimal + number_of_decimal / 3 + (number < 0 ? 1 : 0));

if (number < 0)
	builder.append('-');

auto remainder = abs_number
for (u32 i = number_of_decimal - 1; i > 0 ; --i) {
	builder.append(remainder / pow(10, i));
	if (i != number_of_decimal && i != 0 && i % 3 == 0)
		builder.append(',');
	remainer =  remainer % (pow(10, i) * (remainder / pow(10, i)));
}

return builder.to_deprecated_string();
```

Reply:
Agreed on setting inline capacity in advance, thanks @MacDue.

@LucasChollet do you recommend to go this route? I personally find the vector-based solution more elegant but if optimizing for memory is preferred I can get onboard with this.

Reply:
I don't find the vector-based solution more elegant, so I wouldn't recommend your way. However, the code isn't performance critical, so I'm pretty sure it will be merged anyway.

Calling `ensure_capacity` would be nice tho.